src/chainstate/stacks/boot/mod.rs

```diff
@@ -293,6 +360,107 @@ pub mod test {
 
     use util::hash::to_hex;
 
+    #[test]
+    fn make_reward_set_units() {
+        let threshold = 1_000;
+        let addresses = vec![
+            (
+                StacksAddress::from_string("STVK1K405H6SK9NKJAP32GHYHDJ98MMNP8Y6Z9N0").unwrap(),
+                1500,
+            ),
+            (
+                StacksAddress::from_string("ST76D2FMXZ7D2719PNE4N71KPSX84XCCNCMYC940").unwrap(),
+                500,
+            ),
+            (
+                StacksAddress::from_string("STVK1K405H6SK9NKJAP32GHYHDJ98MMNP8Y6Z9N0").unwrap(),
+                1500,
+            ),
+            (
+                StacksAddress::from_string("ST76D2FMXZ7D2719PNE4N71KPSX84XCCNCMYC940").unwrap(),
+                400,
+            ),
+        ];
+        assert_eq!(
+            StacksChainState::make_reward_set(threshold, addresses).len(),
+            3
+        );
+    }
+
+    #[test]
+    fn get_reward_threshold_units() {
+        // when the liquid amount = the threshold step,
+        //   the threshold should always be the step size.
+        let liquid = POX_THRESHOLD_STEPS_USTX;
+        assert_eq!(
+            StacksChainState::get_reward_threshold(&PoxConstants::new(1000, 1, 1, 1), &[], liquid),
+            POX_THRESHOLD_STEPS_USTX
+        );
+        assert_eq!(
+            StacksChainState::get_reward_threshold(
+                &PoxConstants::new(1000, 1, 1, 1),
+                &[(rand_addr(), liquid)],
+                liquid
+            ),
+            POX_THRESHOLD_STEPS_USTX
+        );
+
+        let liquid = 200_000_000 * MICROSTACKS_PER_STACKS as u128;
+        // with zero participation, should scale to 25% of liquid
+        assert_eq!(
+            StacksChainState::get_reward_threshold(&PoxConstants::new(1000, 1, 1, 1), &[], liquid),
+            50_000 * MICROSTACKS_PER_STACKS as u128
+        );
+        // should be the same at 25% participation
+        assert_eq!(
+            StacksChainState::get_reward_threshold(
+                &PoxConstants::new(1000, 1, 1, 1),
+                &[(rand_addr(), liquid / 4)],
```

Comment:
should this be `liquid / POX_MAXIMAL_SCALING`?

Reply:
I wanted this test to be hardcoded, so that if the POX_MAXIMAL_SCALING is adjusted, this test would need to be adjusted as well.